pyproject.toml

```diff
@@ -34,21 +34,19 @@ types-Deprecated = "^1.2.9"
 pycryptodome = "^3.16.0"
 
 [tool.poetry.dev-dependencies]
-flake8 = "^4.0.1"
-black = "23.3.0"
+flake8 = "^7.0.0"
+black = "24.8.0"
 flake8-black = "^0.3.6"
 flake8-docstrings = "^1.7.0"
 mypy = "^1"
 isort = "^5.11.5"
 flake8-isort = "^6.0.0"
-flake8-annotations = "2.7.0"
 flake8-absolute-import = "^1.0"
 darglint = "^1.5.8"
 sphinx-rtd-theme = "^3.0.0"
 aiounittest = "^1.4.0"
 coverage = "^7.2.7"
 Jinja2 = "^3.1.4"
-MarkupSafe = "2.1.5"
```

Comment:
Why is this removed?

Reply:
I've updated the rationale in the description of the PR.

I couldn't find any usages of this dependency. Where is it used?

Reply:
It's a dependency for Jinja2. I'm not sure why Jinja2 is included though tbh, maybe it was previously a dependency for something.

Reply:
`Jinja2` is also not needed. I'm happy to remove it.

Reply:
If nothing complains about it, sure

Reply:
removed in 563897fb